waves/src/App.tsx

```diff
@@ -0,0 +1,20 @@
+import React, {useState} from 'react';
+import './App.css';
+
+function App() {
+    import('./native/pkg').then(({ hello }) => {
+        let welcome = hello("World");
+        setWelcome(welcome)
+    })
```

Comment:
Can we only use async imports with wasm?

What happens if you try to regularly import this at the top with `import wasm from "./native/pkg"`? Will the wasm get inlined into the JS or is it still loaded as a dedicated resource?

Ideally, we can just use regular imports but the wasm binary is loaded as a dedicated resource.

Potentially relevant: https://webpack.js.org/guides/code-splitting/#dynamic-imports

Reply:
> Can we only use async imports with wasm?

I tried this at first but it wasm imports need to be async.
webpack 5 should solve this issue but it's not yet the default version in create-react-app.

- https://webpack.js.org/configuration/experiments/
- https://github.com/facebook/create-react-app/issues/9994